### DISCUSSION TITLE
Can't install PDM on WSL

### DISCUSSION THREAD

**Discussion by ShaiAvr** [pdm-project/pdm#3205] [Q&A]
I tried to install `pdm` using the installation script for Linux on Ubuntu running under WSL: `curl -sSL https://pdm-project.org/install-pdm.py | python3 -`. The command failed with an error: `urllib.error.URLError: <urlopen error unknown url type: https>`. I use python 3.12.6 through `pyenv`:

![Screenshot 2024-10-11 101547](https://github.com/user-attachments/assets/d6e7d5ec-5e57-4996-8d54-ef8f75d47a8c)

I don't know if this error is specific to WSL, but I can't figure out how to fix it. Any help would be appreciated.

**Comment by TayfurYldz** [comment]
That error is not WSL-specific and it happens before PDM itself is installed. Python's urllib is saying that the interpreter running install-pdm.py has no usable HTTPS handler.

With a normal CPython build, urllib.request supports HTTPS through the ssl module. Run the checks with the exact python3 used in the pipe: import ssl and print ssl.OPENSSL_VERSION, then try urllib.request.urlopen against an HTTPS URL.

If importing ssl fails, fix or rebuild that Python installation rather than changing PDM. Since the report uses a pyenv Python, this commonly means the interpreter was built while the required OpenSSL development libraries were unavailable, or its OpenSSL runtime can no longer be loaded.

A useful isolation test is to run the installer with Ubuntu's system Python explicitly. If /usr/bin/python3 has working ssl while the pyenv 3.12.6 does not, WSL networking and the installer URL are not the root cause; that pyenv interpreter is.

After installing the needed Ubuntu build dependencies/OpenSSL headers, rebuild that Python version with pyenv and verify import ssl before rerunning the PDM installer. I would not work around this by disabling TLS verification: an interpreter that cannot open HTTPS URLs will cause more package-management failures after installation too.
